template/src/main/resources/application.yml

```diff
@@ -1,34 +1,53 @@
-{% for server in asyncapi.servers %}
-  {% if server.scheme === 'amqp' %}
+{%- set hasSubscribe = false -%}
+{%- set hasPublish = false -%}
+{%- for channelName, channel in asyncapi.channels() -%}
+    {%- if channel.hasPublish() -%}
+        {%- set hasPublish = true -%}
+    {%- endif -%}
+    {%- if channel.hasSubscribe() -%}
+        {%- set hasSubscribe = true -%}
+    {%- endif -%}
+{%- endfor -%}
+{%- if asyncapi | isProtocol('amqp') %}
 amqp:
   broker:
     host: localhost
     port: 5672
     username: guest
     password: guest
   exchange:
-{% for topicName, topic in asyncapi.topics %}
-  {% if topic.publish %}
-    {{topic.x-service-name}}: {{topic}}
-  {% endif %}
-{% endfor %}
+  {% for channelName, channel in asyncapi.channels() %}
+    {% if channel.hasSubscribe() and channel.subscribe().bindings('amqp') %}
+    {{channel.subscribe().bindings('amqp').exchange().name()}}: {{channelName}}
+    {% endif %}
+  {% endfor %}
   queue:
-{% for topicName, topic in asyncapi.topics %}
-  {% if topic.subscribe %}
-    {{topic.x-service-name}}: {{topic}}
-  {% endif %}
-{% endfor %}
-  {% endif %}
-  {% if server.scheme === 'mqtt' %}
+  {% for channelName, channel in asyncapi.channels() %}
+    {% if channel.hasPublish() and channel.publish().bindings('amqp') %}
+    {{channel.publish().bindings('amqp').queue().name()}}: {{channelName}}
+    {% endif %}
+  {% endfor %}
+{% endif %}
+{%- if asyncapi | isProtocol('mqtt') %}
+  {% for serverName, server in asyncapi.servers() %}
 mqtt:
   broker:
-    host: tcp://{{server.url}}
-    port: {{server.variables.port.default}}
+    host: tcp://{{server.url()}}
+    port: {{server.variables(port).defaultValue()}}
     username:
     password:
   topic:
-{% for topicName, topic in asyncapi.topics %}
-    {{topic.x-service-name}}Topic: {{topic}}
-{% endfor %}
+    {% for channelName, channel in asyncapi.channels() %}
+    {{channelName}}Topic: {{channelName}}
+    {% endfor %}
+  {% endfor %}
+{% endif %}
+{%- if asyncapi | isProtocol('kafka') %}
+kafka:
+  bootstrap-servers: {% for serverName, server in asyncapi.servers() %}{{server.url()}}{% if not loop.last %},{% endif %}{% endfor %}
+  {%- if hasPublish %}
+  subscribe:
+    pool-timeout: 3000
+    amount-of-listeners: 3
```

Comment:
These just parameters which you could find in [doc for spring-kafka](https://docs.spring.io/spring-kafka/reference/html/#record-listener) 
```
        factory.setConsumerFactory(consumerFactory());
        factory.setConcurrency(3);
        factory.getContainerProperties().setPollTimeout(3000);
```
Sure, i'm agree with your suggestion

Reply:
Custom template configuration needed? here you go https://github.com/asyncapi/generator/blob/master/docs/authoring.md#configuration-file ;) you can define parameters that later one can use during generation to specify if she/he wants `pool-timeout` to be 3000 or 4000. Have a look how html-template is doing it https://github.com/asyncapi/html-template#supported-parameters

Unless, 🤔 the information that you have here is not kafka spring specific but kafka in generat, then maybe it would make sense to specify it in the spec, with spec extension on a server object? 🤔 

Nevertheless, I agree this can be done in a separate PR followed by separate discussion, to not block this PR

Reply:
@derberg thanks for the hint!
I'll think about it and create issue\PR.
But please not here. It couldn't be "one PR for everything". The purpose was to add support of Kafka and update the template to the spec 2.0. My opinion - it's done. 
I do not see any blockers in this PR. Actually this PR is blocker for numerous follow up changes mentioned in "Missing features", comments here e.t.c.
Creating new issues \ PR is a strange activity, as long as we do not have this PR in the master.

Reply:
👍 
just please create an issue so we do not forget

Reply:
@derberg here it is #9